src/ucc/api/ucc.h

```diff
@@ -199,33 +167,260 @@ typedef enum {
  *  Description
  *
  *  @ref ucc_datatype_t represents the datatypes supported by the UCC library’s
- *  collective and reduction operations. The standard operations are signed and
- *  unsigned integers of various sizes, float 16, 32, and 64, and user-defined
- *  datatypes. The UCC_DT_USERDEFINED represents the user-defined datatype. The
- *  UCC_DT_OPAQUE is used to represent the user-defined datatypes for
- *  user-defined reductions. When UCC_DT_OPAQUE is used, the library passes the
- *  data to the user-defined reductions without any modifications.
+ *  collective and reduction operations. The standard operations (predefined)
+ *  are signed and unsigned integers of various sizes, float 16, 32, and 64, and
+ *  user-defined datatypes. User-defined datatypes cab created using
+ *  @ref ucc_dt_create_generic call and can support user-defined reduction
+ *  operations. Standard (predefined) reduction operations can be used only with
```

Comment:
cab? Typo? "User-defined datatypes are created using @ref ucc_dt_create generic interfaces. The User-defined datatypes can be used for both predefined and user-defined reduction operations, while predefined datatypes can only be used for predefined reduction operations."

Reply:
@manjugv plz note, with this proposal if user wants CUSTOM reduction even on pre-defined dtype he will still need to wrap the simple dtype into "generic". This is some additional actions from user (w/o perf penalty though since it is light-weight) but it simplifies internal dt/op checks logic - this is why we went with this proposal.

Reply:
So, typo fixed. the description is still basically the same.